registry/remote/example_test.go

```diff
@@ -0,0 +1,542 @@
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	descriptor, err := repo.Blobs().Resolve(ctx, digest.FromBytes([]byte(exampleLayer)).String())
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+
+	// option 1: fetch a WHOLE blob
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// option 2: partially read, if the remote registry supports
+	if seeker, ok := r.(io.ReadSeeker); ok {
+		offset := int64(8)
+		_, err = seeker.Seek(offset, io.SeekStart)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		pulledBlob, err := io.ReadAll(r)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		// caller SHOULD verify what is fetched
+		if descriptor.Size-offset != int64(len(pulledBlob)) {
+			panic(err) // Handle error
+		}
+		fmt.Println(string(pulledBlob))
+	}
+
+	// Output:
+	// Example layer content
+	// layer content
+}
+
+// ExampleRepository_Tag gives example snippets for tagging a manifest.
+func ExampleRepository_Tag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// Tag a manifest with below digest
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	err = repo.Tag(ctx, descriptor, tag)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println("Succeed")
+
+	// Output:
+	// Succeed
+}
+
+// ExampleRegistry_Repositories gives example snippets for listing respositories in a HTTPS registry with pagination.
+func ExampleRegistry_Repositories() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// Override the `host` variable to play with local registry.
+	// Uncomment below line to reset HTTP option:
+	// reg.PlainHTTP = true
+	ctx := context.Background()
+	err = reg.Repositories(ctx, func(repos []string) error {
+		for _, repo := range repos {
+			fmt.Println(repo)
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// Output:
+	// public/repo1
+	// public/repo2
+	// internal/repo3
+}
+
+func Example_pullByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
```

Comment:
Same as above probably can remove the comment and there is no `GET` operation is my understanding. 
Try to see if you find all occurences of this pattern and remove the comment.